Add `replit.app`

lincoln-replit opened the pull request:
Public Suffix List (PSL) Pull Request (PR) Template
====

Each PSL PR needs to have a description, rationale, indication of DNS validation and syntax checking, as well as a number of acknowledgements from the submitter.  This template must be included with each PR, and the submitting party MUST provide responses to all of the elements in order to be considered.

<!-- #### READ THIS FIRST ####

If you haven't yet, please read our guidelines:
https://github.com/publicsuffix/list/wiki/Guidelines#submit-the-change

Also, read them again, as many skip that part and 
get confused about why their PR is delayed or does
not get accepted when theirs didn't follow them.

A recent PR using the current template is 
https://github.com/publicsuffix/list/pull/1591, although 
the organization and description were not as substantial 
as desired, which required maintainers time to visit the 
requestors website to further research. 
Having more robust org/desc improves the PR processing 
pace due to the extra cycles not lost to research.
For an example of what an excellent description in a PR looks like
see https://github.com/publicsuffix/list/pull/615, 
although that example uses an earlier template.
-->
### Checklist of required steps

* [ ] Description of Organization
* [ ] Robust Reason for PSL Inclusion
* [ ] DNS verification via dig
* [ ] Run Syntax Checker (make test)

* [ ] Each domain listed in the PRIVATE section has and shall maintain at least two years remaining on registration, and we shall keep the \_PSL txt record in place in the respective zone(s) in the affected section

__Submitter affirms the following:__ 
<!--
Third-party Limits are used elsewhere, such as at Cloudflare, Let's 
Encrypt, Apple, GitLab or others, and having an entry in the PSL alters 
the manner in which those third-party systems or products treat 
a given domain name or sub-domains within it.

To be clear, it is appropriate to address how those limits impact 
your domain(s) directly with that third-party, and it is inappropriate 
to submit entries to the PSL as a means to work around those limits or 
restrictions.
-->
  * [ ] We are listing *any* third-party limits that we seek to work around in our rationale such as those between IOS 14.5+ and Facebook (see [Issue #1245](https://github.com/publicsuffix/list/issues/1245) as a well-documented example)

<!--
The purpose of the question above is to expose limit workarounds.
If there are third party limits that the PR seeks to overcome, those
must be listed within the rationale section of this request, and 
provide a good level of detail the effort that was made to work directly 
with the third part(y|ies) in attempting to address this within their 
rationale response below.
In all cases, software and services should be discouraged from use of
the PSL as a rate-limiting tool, and provide clear instructions to their
own clients, partners and users on the manner in which they can directly
request rate limit increases.
We treat the following as an attestation in the public record of the 
requesting party that they are not attempting to bypass rate limits through
the PR.
-->

  * [ ] This request was _not_ submitted with the objective of working around other third-party limits

<!--
The guidelines describe which section to place the entry, what the 
order of commented org placement, order of sorting of entries. 
(hint: TLD then SLD, Ascending sort)   Although it seems pedantic, 
the sorting and formatting rules help ensure all of the automation 
that uses the PSL operates correctly.  Typically both are solved or
neither.
-->

  * [ ] The [Guidelines](https://github.com/publicsuffix/list/wiki/Guidelines) were carefully _read_ and _understood_, and this request conforms
  * [ ] The submission follows the [guidelines](https://github.com/publicsuffix/list/wiki/Format) on formatting and sorting

<!-- 
Sorting and formatting of the entries is outlined in the guidelines 
and non-conforming requests are one of the largest sources of delay,
so getting this right initially will aid successfully having it 
proceed.  Miss-located entries and trailing spaces should be avoided.
-->

---

For Private section requests that are submitting entries for domains that match their organization website's primary domain, please understand that this can have impacts that may not match the desired outcome and take a long time to rollback, if at all.

To ensure that requested changes are entirely intentional, make sure that you read the affectation and propagation expectations, that you understand them, and confirm this understanding. 

PR Rollbacks have lower priority, and the volunteers are unable to control when or if browsers or other parties using the PSL will refresh or update.

<!-- 
Seriously, carefully read the downline flow of the PSL and the 
guidelines. Your request could very likely alter the cookie and 
certificate (as well as other) behaviours on your core domain name in 
ways that could be problematic for your business.

Rollback is really not predictable, as those who use or incorporate 
the PSL do what they do, and when. It is not within the PSL volunteers' 
control to do anything about that.  

The volunteers are busy with new requests, and rollbacks are lowest 
priority, so if something gets broken by your PR, it will potentially 
stay that way for an indefinite period of time (typically long).
-->

(Link: [about propagation/expectations](https://github.com/publicsuffix/list/wiki/Guidelines#appropriate-expectations-on-derivative-propagation-use-or-inclusion))

 * [ ] *Yes, I understand*.  I could break my organization's website cookies etc. and the rollback timing, etc is acceptable.  *Proceed*.
---


<!--

As you complete each item in the checklist please mark it with an X

Example:

* [x] Description of Organization

-->

Description of Organization
====

Replit is an online programming environment that allows users to create and host their own websites and applications. I am an engineer at Replit.

In order to continue to fight spam and abuse on our platform, we are launching a new domain to host user websites. It will behave similarly to the `repl.co` domain where Replit users' websites are hosted now, but there will be stricter requirements around who can deploy to this domain.

Organization Website: https://replit.com/

Reason for PSL Inclusion
====

Replit users host their websites on replit.app, just like they do on repl.co. User websites should be isolated and cookies should be properly segregated.

We have held the `replit.app` domain for some time and will continue to hold it forever.

Number of users this request is being made to serve: over 1 million


DNS Verification via dig
=======

```
$ dig +short TXT _psl.replit.app

$ dig +short TXT _psl.id.replit.app

```

<!--
For each domain you'd like to add to the list please create
a DNS verification record pointing to your pull request.

For example, if you'd like to add example.com and example.net
you would need to provide the following verifications:

```
dig +short TXT _psl.example.com
"https://github.com/publicsuffix/list/pull/XXXX"
```

```
dig +short TXT _psl.example.net
"https://github.com/publicsuffix/list/pull/XXXX"
```

Note that XXXX is replaced with the number of your pull request.

We ask that you leave this record in place while you want 
your entry to remain in the PSL, so that future (TBD) 
automation can remove entries where the record is not present.

-->

Results of Syntax Checker (`make test`)
=========

`make test` passed.

dnsguru commented:
Also see:   #1239 #1568

lincoln-replit commented:
Is there anything else we need to do to get this added to the list? Happy to provide any other documentation that would help to verify this domain.

dnsguru reviewed (changes requested):
Inches from goal, I _finally_ had some volunteer cycles to put in to the project and was working to advance this pull request when I noticed that the sort order doesnt match the guidelines...  

Please sort these by TLD,SLD,3LD ascending and @ me when you do it  (ie  the .app ones up above the .co, the .dev after the .co and .run at the end)  The third levels are in the right order within their sld.tld (id.foo.bar after foo.bar), but the TLDs just need a quick move.

lincoln-replit commented:
Thanks @dnsguru -- I believe I've got the sorting correct now.

lincoln-replit commented:
FYI, I added a couple more subdomains to the list here. All should be configured and alphabetized as expected and I've updated the PR description as well.

lincoln-replit commented:
To clarify @mozfreddyb, are you saying that any additional work needs to be done on our part?

simon-friedberger reviewed (changes requested):
* [ ] Expiration
  * [ ] replit.app expires 2025-04-16
  * [x] replit.dev expires 2029-01-26
* [x] DNS _psl entries
* [x] Tests pass
* [x] Sorting
* [x] Reasoning/Organization description

simon-friedberger commented on public_suffix_list.dat:
```diff
@@ -14172,10 +14172,39 @@ onrender.com
 
 // Repl.it : https://repl.it
 // Submitted by Lincoln Bergeson <lincoln@replit.com>
```
Please replace this with a non-personal address.

lincoln-replit replied:
Done

lincoln-replit commented:
Sorry about that, I thought the registration was 5+ yrs out. That's fixed now.

simon-friedberger commented:
```
$ dig +short TXT _psl.replit.dev
"https://github.com/publicsuffix/list/pull/1679"
```
Did this disappear? Please keep the >2y validity and the DNS entries for as long as you want to keep the domains on the PSL!

lincoln-replit commented:
@simon-friedberger Just added it back in! We'll make sure to keep it on there.
